src/compiler/checker.ts

```diff
@@ -632,7 +632,10 @@ namespace ts {
         }
 
         function mergeSymbol(target: Symbol, source: Symbol) {
-            if (!(target.flags & getExcludedSymbolFlags(source.flags))) {
+            if (!(target.flags & getExcludedSymbolFlags(source.flags)) ||
+                source.valueDeclaration && source.valueDeclaration.kind === SyntaxKind.Identifier ||

```

Comment:
I'd say that there should _probably_ be an extra symbol flag you set on these symbols to imply this, rather than relying on the value declaration being an identifier implying this.

Reply:
Good idea, for now at least. There are still symbol flags to burn.

Reply:
Done.